finagle-redis/src/main/scala/com/twitter/finagle/redis/SentinelCommands.scala

```diff
@@ -0,0 +1,141 @@
+package com.twitter.finagle.redis
+
+import _root_.java.lang.{ Long => JLong, Boolean => JBoolean }
+import scala.collection.immutable.{ Set => ImmutableSet }
+import com.twitter.finagle.redis.protocol._
+import com.twitter.finagle.redis.protocol.commands._
+import com.twitter.finagle.redis.util._
+import com.twitter.util.Future
+import org.jboss.netty.buffer.ChannelBuffer
+import java.net.InetAddress
+import java.net.InetSocketAddress
+
+trait SentinelCommands { self: BaseClient =>
+
+  import SentinelClient._
+
+  /**
+   * Show a list of monitored masters and their state.
+   */
+  def masters(): Future[List[MasterNode]] =
+    doRequest(SentinelMasters()) {
+      case MBulkReply(masters) => Future.value(
+        masters map {
+          case MBulkReply(messages) => new MasterNode(
+            returnMap(ReplyFormat.toChannelBuffers(messages)))
+        })
+      case EmptyMBulkReply() => Future.value(Nil)
+    }
+
+  /**
+   * Show the state and info of the specified master.
+   */
+  def master(name: ChannelBuffer): Future[MasterNode] =
+    doRequest(SentinelMaster(name)) {
+      case MBulkReply(messages) => Future.value(
```

Comment:
is it possible that there's no master, for example if the master just died and the dead master has been detected, but leader election hasn't finished yet?